test/reusing-processes/index.js

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const assert = require('assert');
+const { resolve } = require('path');
+const { exec } = require('child_process');
+const libExecutable = resolve(__dirname, '../../dist/bin/cli.js');
+
+exec(`${libExecutable} --max-parallel 1 test/reusing-processes/index.spec.js test/reusing-processes/index.2.spec.js`, {
+    cwd: resolve(__dirname, '../../'),
+}, (err, stdout) => {
+    if (err) {
+        console.error(err);
+        process.exit(1);
+    }
+
+    const [pid1, pid2] = stdout.split(/\n/g).filter(line => /^pid/.test(line));
+
+    if (pid1 && pid2) assert.strictEqual(pid1, pid2);
+});
```

Comment:
Thanks for adding the tests. `mocha-parallel-tests` is highly covered with tests itself and that's why it's so stable. This PR, however, introduces a very important piece of logic and I'm thinking about at least two additional tests that would make sense here:

1) Let's assume that we have asynchronous test suite A which is executing for 3 seconds, asynchronous test suite B which is executing for 2 seconds and synchronous test case C; `--max-parallel` is equal to 2. For this configuration I expect test case C to re-use the process from test case B, while A will be still running in its own process.
2) The test case from #155: what should be the behaviour when the forked process is blocked from exiting (or running the next suite?) by some code (for instance it can launch the http server in "before" hook). I assume we will need to forget about this process and fork new ones but maybe you have a different idea? The reason why we need this is because `mocha-parallel-tests` is trying to act exactly like `mocha`. When it can do this it's a shocking surprise for our users.

Could you please also add these tests and maybe you can think of any others?

Reply:
1. Ok I'll try to write that test
2. if some test hoards the process, then I suppose it eventually would time out. I don't think I've implemented that. I'll do it now.

Reply:
1. Could you please point me to the test you've written?

What's the CP?